### DISCUSSION TITLE
Tailscale fails to start after install

### DISCUSSION THREAD

**Discussion by yodaphone** [tteck/Proxmox#814] [Help Needed] [answered]
Ran the Tailscale script and it asks the question have you replaced it with your LXC ID & i say Y. the script says " Installing Tailscale...
" and immediately ends. No additional messages etc.

The LXC is created on PVE but when i try to start it, i get this error

TASK ERROR: missing 'arch' - internal error at /usr/share/perl5/PVE/LXC.pm line 607.

**Comment by tteck** [comment, marked as answer]
This script will add Tailscale to an **existing LXC Container ONLY**

**Comment by buczi11** [comment]
What node are you adding Tailscale to?

**Reply by yodaphone** [reply]
Never mind. I was under the assumption that it will create the LXC and install. It worked when I ran it after I created the LXC

**Comment by mrtrofimov** [comment]
Hi guys! I got stuck with an installation problem. I have a newly created LXC container that I created with a debian 11 template. When I run the script, script says "Installing Tailscale..." and immediately ends. No additional messages etc. Tell me what I'm doing wrong. Sorry I'm new to this.

**Reply by tteck** [reply]
The script runs without any problems from this end. 

```
  ______      _ __                __        
 /_  __/___ _(_) /_____________ _/ /__      
  / / / __ `/ / / ___/ ___/ __ `/ / _ \     
 / / / /_/ / / (__  ) /__/ /_/ / /  __/     
/_/  \__,_/_/_/____/\___/\__,_/_/\___/      
                                            
This script will add Tailscale to an existing LXC Container ONLY
Did you replace 106 with your LXC ID? Proceed(y/n)?y
Installing Tailscale...
Installed Tailscale
 ✔ Completed Successfully!
 Reboot 111 LXC to apply the changes, then run tailscale up in the LXC console
```

**Reply by tteck** [reply]
Are you using the Proxmox shell or SSH to run the script?

**Reply by mrtrofimov** [reply]
I use Proxmox shell

**Reply by tteck** [reply]
Did you create the Debian 11 LXC manually, or did you use the script for that?

**Reply by mrtrofimov** [reply]
I did it manually

**Reply by tteck** [reply]
You should use the script, specifically the "Advanced settings," to create a Debian 11 LXC and see if you encounter the same problem.

**Reply by mrtrofimov** [reply]
I'll try and come back. Thank you.

**Reply by mrtrofimov** [reply]
@tteck Today I've upgraded my Proxmox from 7 to 8 version using your scripts. Everything is Okay. 
Then the new LXC Debian 12 was created by your script as well. 
But when I'm installing Tailscale I got the same problem:
<img width="582" alt="Снимок экрана 2023-09-02 в 15 26 45" src="https://github.com/tteck/Proxmox/assets/118134300/557b0aed-e629-4b3b-baef-72e85f6c1c03">

I've got the error in the tasks log: 
failed waiting for client: timed out
TASK ERROR: command '/usr/bin/termproxy 5900 --path /vms/102 --perm VM.Console -- /usr/bin/dtach -A /var/run/dtach/vzctlconsole102 -r winch -z lxc-console -n 102 -e -1' failed: exit code 1

**Reply by tteck** [reply]
The script runs without any problems from this end. 
I created a new Debian 12 LXC
```
    ____       __    _           
   / __ \___  / /_  (_)___ _____ 
  / / / / _ \/ __ \/ / __ `/ __ \
 / /_/ /  __/ /_/ / / /_/ / / / /
/_____/\___/_.___/_/\__,_/_/ /_/ 
                                 
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 112
Using Hostname: debian-tailscale
Using Disk Size: 2
Allocated Cores: 1
Allocated RAM: 512
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Debian LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 112 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.253 
 ✓ Internet Connected
 ✓ DNS Resolved github.com to 140.82.114.4
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Customized Container
 ✓ Cleaned
 ✓ Completed Successfully!
```
Then added Tailscale
```
  ______      _ __                __        
 /_  __/___ _(_) /_____________ _/ /__      
  / / / __ `/ / / ___/ ___/ __ `/ / _ \     
 / / / /_/ / / (__  ) /__/ /_/ / /  __/     
/_/  \__,_/_/_/____/\___/\__,_/_/\___/      
                                            
This script will add Tailscale to an existing LXC Container ONLY
Did you replace 106 with your LXC ID? Proceed(y/n)?y
Installing Tailscale...
Installed Tailscale
 ✔ Completed Successfully!
 Reboot 112 LXC to apply the changes, then run tailscale up in the LXC console
root@server:~# 
```
No issues 🤷🏻‍♂️

**Reply by tteck** [reply]
I believe it relates to the configuration or setup of your system.

Since I've adopted the 'Keep It Simple, Stupid' (K.I.S.S.) principle, I've set up my system with the ext4 filesystem and established containers/virtual machines using local LVM storage. This approach tends to result in fewer issues or problems.

**Reply by mrtrofimov** [reply]
If I only could guess what the problem is. I tried to do the same with LXC Ubuntu with the same result. What kind of a diagnostic information could help?

**Reply by JuanGrados** [reply]
@mrtrofimov did you succeed? I have the same problem, I tried with two different LXC, a Debian based and Ubuntu based.
Then running the script from the shell of the PEV and nothing, it just exits after "Installing Tailscale...".
If I try to run `tailscale up` from the LXC it says command not found...

**Reply by tteck** [reply]
Just checked again

```
  ______      _ __                __        
 /_  __/___ _(_) /_____________ _/ /__      
  / / / __ `/ / / ___/ ___/ __ `/ / _ \     
 / / / /_/ / / (__  ) /__/ /_/ / /  __/     
/_/  \__,_/_/_/____/\___/\__,_/_/\___/      
                                            
This will add Tailscale to an existing LXC Container ONLY. Proceed(y/n)?y
Installing Tailscale...
Installed Tailscale
 ✔ Completed Successfully!
 Reboot 111 LXC to apply the changes, then run tailscale up in the LXC console
root@server:~# 
```
Are you using SSH?

**Reply by JuanGrados** [reply]
Thanks for checking.
No, I am not running via ssh, I am doing it from the web interface of Proxmox.
However, after restarting the server it is running just fine.

Thanks again. 
cheers

**Reply by samsong** [reply]
if anyone else has this problem:
* I had installed debian 12 standard via proxmox template. same issue "command not found" & the installer finished without any message.

* deleted the container. created again using ttecks own lxc ubuntu script. version 24.

* then run the command again and it works. must be a problem with the debian lxc i created. not sure why
